src/hooks/use-fetch-collectible-attributes.hook.ts

```diff
@@ -0,0 +1,44 @@
+import { isNonEmptyArray } from '@apollo/client/utilities';
+import { useEffect, useState } from 'react';
+import { EMPTY } from 'rxjs';
+import { map, tap, finalize, catchError } from 'rxjs/operators';
+
+import { CollectibleAttributes } from '../apis/objkt/types';
+import { BLURED_COLLECTIBLE_ATTRIBUTE_NAME } from '../modals/collectible-modal/constants';
+import { CollectibleInterface } from '../token/interfaces/collectible-interfaces.interface';
+import { getAttributesInfo$, getAttributesWithRarity } from '../utils/collectibles.utils';
+
+export const useFetchCollectibleAttributes = (collectible: CollectibleInterface) => {
```

Comment:
I think this logic should be in epic, not in hook

Reply:
We don't need to move it in epic, because we don't get rarity from Objkt. We fetch these attributes data only on CollectibleModal for each collectible, and we don't save this info to store because we have Interval for updating details data, and this rarity field will be overwritten